lib/node_modules/@stdlib/lapack/base/dlacpy/lib/base.js

```diff
@@ -0,0 +1,132 @@
+* @param {integer} strideB1 - stride of the first dimension of `B`
+* @param {integer} strideB2 - stride of the second dimension of `B`
+* @param {PositiveInteger} offsetB - starting index for `B`
+* @returns {Float64Array} `B`
+*
+* @example
+* var Float64Array = require( '@stdlib/array/float64' );
+*
+* var A = new Float64Array( [ 1.0, 2.0, 3.0, 4.0 ] );
+* var B = new Float64Array( 4 );
+*
+* dlacpy( 'row-major', 'all', 2, 2, A, 2, 1, 0, B, 2, 1, 0 );
+* // B => <Float64Array>[ 1.0, 2.0, 3.0, 4.0 ]
+*/
+function dlacpy( order, uplo, M, N, A, strideA1, strideA2, offsetA, B, strideB1, strideB2, offsetB ) { // eslint-disable-line max-len, max-params
+	var sa0;
+	var sa1;
+	var sb0;
+	var sb1;
+	var i0;
+	var i1;
+
+	if ( order === 'row-major' ) {
+		// For row-major matrices, the last dimension has the fastest changing index...
+		sa0 = strideA2; // stride for innermost loop
+		sa1 = strideA1; // stride for outermost loop
+		sb0 = strideB2; // stride for innermost loop
+		sb1 = strideB1; // stride for outermost loop
+	} else { // order === 'column-major'
+		// For column-major matrices, the first dimension has the fastest changing index...
+		sa0 = strideA1; // stride for innermost loop
+		sa1 = strideA2; // stride for outermost loop
+		sb0 = strideB1; // stride for innermost loop
+		sb1 = strideB2; // stride for outermost loop
+	}
+
+	if ( uplo === 'upper' ) {
+		if ( order === 'column-major' ) {
+			for ( i1 = 0; i1 < N; i1++ ) {
+				for ( i0 = 0; i0 < min( i1, M ); i0++ ) {
+					B[ offsetB + ( i0 * sb0 ) + ( sb1 * i1 ) ] = A[ offsetA + ( i0 * sa0 ) + ( sa1 * i1 ) ]; // eslint-disable-line max-len
+				}
+			}
+			return B;
+		}
+		// order === 'row-major'
+		for ( i1 = 0; i1 < N; i1++ ) {
+			for ( i0 = 0; i0 < min( i1, M ); i0++ ) {
+				B[ offsetB + ( i0 * sb1 ) + ( i1 * sb0 ) ] = A[ offsetA + ( i0 * sa1 ) + ( i1 * sa0 )]; // eslint-disable-line max-len
+			}
+		}
+		return B;
+	}
+	if ( uplo === 'lower' ) {
+		if ( order === 'column-major' ) {
+			for ( i1 = 0; i1 < N; i1++ ) {
+				for ( i0 = i1; i0 < M; i0++ ) {
+					B[ offsetB + ( i0 * sb0 ) + ( sb1 * i1 ) ] = A[ offsetA + ( i0 * sa0 ) + ( sa1 * i1 ) ]; // eslint-disable-line max-len
+				}
+			}
+			return B;
+		}
+		// order === 'row-major'
+		for ( i1 = 0; i1 < N; i1++ ) {
+			for ( i0 = i1; i0 < M; i0++ ) {
+				B[ offsetB + ( i0 * sb1 ) + ( i1 * sb0 ) ] = A[ offsetA + ( i0 * sa1 ) + ( i1 * sa0 )]; // eslint-disable-line max-len
+			}
+		}
+		return B;
+	}
+	// All of the matrix `A`
+	if ( order === 'column-major' ) {
+		for ( i1 = 0; i1 < N; i1++ ) {
+			for ( i0 = 0; i0 < M; i0++ ) {
+				B[ offsetB + ( i0 * sb0 ) + ( sb1 * i1 ) ] = A[ offsetA + ( i0 * sa0 ) + ( sa1 * i1 ) ]; // eslint-disable-line max-len
+			}
+		}
+		return B;
+	}
+	// order === 'row-major'
+	for ( i1 = 0; i1 < N; i1++ ) {
+		for ( i0 = 0; i0 < M; i0++ ) {
+			B[ offsetB + ( i0 * sb1 ) + ( i1 * sb0 ) ] = A[ offsetA + ( i0 * sa1 ) + ( i1 * sa0 )]; // eslint-disable-line max-len
+		}
+	}
```

Comment:
This looks off to me. Loop interchange also involves swapping `N` and `M` (if necessary). Once done, then you don't need separate loops for row- and column-major. See https://github.com/stdlib-js/stdlib/blob/9abdead65457225ccbf44f4bab8caf96167a71c5/lib/node_modules/%40stdlib/ndarray/base/assign/lib/2d.js#L108.

Reply:
I referred https://github.com/OpenMathLib/OpenBLAS/blob/6013b36b164b35821930e46db12f7388e6db9adc/lapack-netlib/LAPACKE/src/lapacke_dlacpy_work.c#L71-L77 and we need to do a transpose of the matrices to perform the swap. I am not sure how to support a transpose operation here.